### PR TITLE
Txs by blockid to store them in postrress

### PR DESCRIPTION
**Pull request by lmahanand**
## Problem
Currently locked and redeem transactions are directly being retrieved from blockchain. Instead we need them to be getting stored in postgress db so that we can have improved performance.
Asana Link: 

## Solution
Add a tx-type field to transaction table.
Implementation to get the locked and redeem transaction directly from postgress db.
## Testing Done and Results
Tested locally.

### CODE REVIEW

File: store/postgres/postgres.go [hunk 1]
```diff
@@ -156,6 +156,28 @@ WHERE
...
+	locked_amount,
+	nonce,
+	message,
+	sign,
+	status,
+	block_id,
+	created_date
+FROM "transaction"
+WHERE
+block_id = $1
```
Comment (cuonglm) [inline store/postgres/postgres.go]: Missing `block_id = $2 AND type = $1`,

File: store/migration/00005_tx_type.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+	"github.com/pressly/goose"
+)
+
+func init() {
+	goose.AddMigration(Up00005, Down00005)
+}
+
+func Up00005(tx *sql.Tx) error {
+	if _, err := tx.Exec(`ALTER TABLE transaction ADD COLUMN tx_type TYPE VARCHAR(64)`); err != nil {
```
[the remark this answers is not in the document]
Comment (lmahanand) [reply store/migration/00005_tx_type.go]: Won't it be conflicting with the keyword **TYPE** in postgress query?

[the remark this answers is not in the document]
Comment (cuonglm) [reply store/migration/00005_tx_type.go]: @lmahanand we can prevent it by double quote `"type"`, but I think we can keep it this way.

[the remark this answers is not in the document]
Comment (lmahanand) [reply store/migration/00005_tx_type.go]: Enclosed **type** in double quotes to stop it being interpreted as a keyword

File: store/migration/00005_tx_type.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+	"github.com/pressly/goose"
+)
+
+func init() {
+	goose.AddMigration(Up00005, Down00005)
+}
+
+func Up00005(tx *sql.Tx) error {
+	if _, err := tx.Exec(`ALTER TABLE transaction ADD COLUMN "type" TYPE VARCHAR(64)`); err != nil {
```
Comment (cuonglm) [inline store/migration/00005_tx_type.go]: Should add an index

File: store/tx.go [hunk 4]
```diff
@@ -26,6 +26,7 @@ type Tx struct {
 	Status          string    `db:"status"`
 	BlockID         uint64    `db:"block_id"`
 	CreationDT      time.Time `db:"created_date"`
+	TxType          string    `db:"tx_type"`
```
[the remark this answers is not in the document]
Comment (cuonglm) [reply store/tx.go]: Type          string    `db:"type"`

File: store/postgres/postgres.go [hunk 5]
```diff
@@ -156,6 +156,28 @@ WHERE
...
+	locked_amount,
+	nonce,
+	message,
+	sign,
+	status,
+	block_id,
+	created_date
+FROM "transaction"
+WHERE
+block_id = $2 AND type = $1
```
Comment (cuonglm) [inline store/postgres/postgres.go]: Since you created lower index, this comparison must be done in lower.

**Pull request merged**